go/inst/instance_dao.go

```diff
@@ -486,16 +530,39 @@ func ReadTopologyInstanceBufferable(instanceKey *InstanceKey, bufferWrites bool,
 			go func() {
 				defer waitGroup.Done()
 				err := sqlutils.QueryRowsMap(db, "show global status like 'rpl_semi_sync_%'", func(m sqlutils.RowMap) error {
-					if m.GetString("Variable_name") == "Rpl_semi_sync_master_status" {
+					variableName := m.GetString("Variable_name")
+					matched, regexperr := regexp.MatchString("^Rpl_semi_sync_(master|source)_status$", variableName)
+					if regexperr != nil {
+						return regexperr
+					}
+					if matched {
 						instance.SemiSyncMasterStatus = (m.GetString("Value") == "ON")
-					} else if m.GetString("Variable_name") == "Rpl_semi_sync_master_clients" {
+						return nil
+					}
+
+					matched, regexperr = regexp.MatchString("^Rpl_semi_sync_(master|source)_clients$", variableName)
+					if regexperr != nil {
+						return regexperr
+					}
+					if matched {
 						instance.SemiSyncMasterClients = m.GetUint("Value")
-					} else if m.GetString("Variable_name") == "Rpl_semi_sync_slave_status" {
-						instance.SemiSyncReplicaStatus = (m.GetString("Value") == "ON")
+						return nil
 					}
 
+					matched, regexperr = regexp.MatchString("^Rpl_semi_sync_(slave|replica)_status$", variableName)
+					if regexperr != nil {
+						return regexperr
+					}
+					if matched {
+						instance.SemiSyncReplicaStatus = (m.GetString("Value") == "ON")
+					}
 					return nil
 				})
+				/*
+					log.Debugf("%v, SemiSyncMasterStatus: %v", instance.Key, instance.SemiSyncMasterStatus)
+					log.Debugf("%v, SemiSyncMasterClients: %v", instance.Key, instance.SemiSyncMasterClients)
+					log.Debugf("%v, SemiSyncReplicaStatus: %v", instance.Key, instance.SemiSyncReplicaStatus)
+				*/
```

Comment:
Apart from these debug logs, LGTM.

It's definitely looks more clear than what I imagined initially 👍